<<<<<<< HEAD:package/python3/0013-Add-option-to-disable-the-sqlite3-module.patch
From dd2722dec08eb9c72c36313e93661eeca3ad64d8 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:20:45 -0800
=======
From a24aee2ba81617baf01215b333c26434355bf91a Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
Date: Tue, 6 Feb 2024 22:12:20 +0100
>>>>>>> upstream-2025.02.7:package/python3/0007-Add-option-to-disable-the-sqlite3-module.patch
Subject: [PATCH] Add option to disable the sqlite3 module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@bootlin.com>
---
<<<<<<< HEAD:package/python3/0013-Add-option-to-disable-the-sqlite3-module.patch
 Makefile.pre.in | 7 +++++--
 configure.ac    | 9 +++++++++
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 5847029..5628860 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1446,7 +1446,6 @@ LIBSUBDIRS=	asyncio \
 		logging \
=======
 Makefile.pre.in | 5 ++++-
 configure.ac    | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index aceb52e31ce..7c5bcea59ba 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2108,7 +2108,6 @@ LIBSUBDIRS=	asyncio \
>>>>>>> upstream-2025.02.7:package/python3/0007-Add-option-to-disable-the-sqlite3-module.patch
 		multiprocessing multiprocessing/dummy \
 		site-packages \
-		sqlite3 \
 		tkinter \
 		turtledemo \
<<<<<<< HEAD:package/python3/0013-Add-option-to-disable-the-sqlite3-module.patch
 		unittest \
@@ -1459,7 +1458,6 @@ LIBSUBDIRS=	asyncio \
 TESTSUBDIRS=	ctypes/test \
 		distutils/tests \
 		idlelib/idle_test \
-		sqlite3/test \
 		test test/audiodata \
 		test/capath test/cjkencodings \
 		test/data test/decimaltestdata \
@@ -1540,6 +1538,11 @@ TESTSUBDIRS += lib2to3/tests			\
 	lib2to3/tests/data/fixers/myfixes
=======
@@ -2278,6 +2277,10 @@ ifeq (@IDLE@,yes)
 LIBSUBDIRS += idlelib idlelib/Icons
>>>>>>> upstream-2025.02.7:package/python3/0007-Add-option-to-disable-the-sqlite3-module.patch
 endif
 
+ifeq (@SQLITE3@,yes)
+LIBSUBDIRS += sqlite3
+TESTSUBDIRS += sqlite3/test
+endif
+
 TEST_MODULES=@TEST_MODULES@
<<<<<<< HEAD:package/python3/0013-Add-option-to-disable-the-sqlite3-module.patch
 libinstall:	build_all $(srcdir)/Modules/xxmodule.c
 	@for i in $(SCRIPTDIR) $(LIBDEST); \
diff --git a/configure.ac b/configure.ac
index 34c2ba9..dfee472 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3373,6 +3373,15 @@ if test "$posix_threads" = "yes"; then
       AC_CHECK_FUNCS(pthread_getcpuclockid)
 fi
=======
 
 .PHONY: libinstall
diff --git a/configure.ac b/configure.ac
index ab68cc3dd73..3fbcabe29c8 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4519,6 +4519,13 @@ AS_VAR_IF([posix_threads], [stub], [
   AC_DEFINE([HAVE_PTHREAD_STUBS], [1], [Define if platform requires stubbed pthreads support])
 ])
>>>>>>> upstream-2025.02.7:package/python3/0007-Add-option-to-disable-the-sqlite3-module.patch
 
+AC_SUBST(SQLITE3)
+AC_ARG_ENABLE(sqlite3,
+	AS_HELP_STRING([--disable-sqlite3], [disable sqlite3]),
+	[ SQLITE3="${enableval}" ], [ SQLITE3=yes ])
+AS_IF([test "$SQLITE3" = "no"],
+      [PY_STDLIB_MOD_SET_NA([_sqlite3])])
+
 AC_SUBST(PYDOC)
 
 AC_ARG_ENABLE(pydoc,
-- 
<<<<<<< HEAD:package/python3/0013-Add-option-to-disable-the-sqlite3-module.patch
2.30.2
=======
2.39.5
>>>>>>> upstream-2025.02.7:package/python3/0007-Add-option-to-disable-the-sqlite3-module.patch

